client/my-sites/plan-features-2023-grid/components/features.tsx

```diff
@@ -28,12 +29,34 @@ const FreePlanCustomDomainFeature: React.FC< { domainName: string } > = ( { doma
 		isError,
 	} = DomainSuggestions.useGetWordPressSubdomain( domainName );
 
+	const translate = useTranslate();
+	const isCustomDomainAllowedOnFreePlan = useIsCustomDomainAllowedOnFreePlan( domainName );
+	const wpcomFreeDomain = ! isError && wordPressSubdomainSuggestions?.[ 0 ]?.domain_name;
```

Comment:
I think these suggestions should really be passed into the plans-features-2023-grid components from outside, either as a list of suggestions or a hook to call. not fond of the network dependency that's hidden inside here.

Reply:
Maybe the domain 'feature' should be injected as its own component and it can then simply be rendered on the grid 🤔 

The domain feature is going to be experimented heavily as suggested by this PR. And this could be the approach we decouple the grid from external dependencies when they are required to be experimented on.

Reply:
Good suggestion! I've addressed this in a separate PR so it's better scoped: https://github.com/Automattic/wp-calypso/pull/79678 and I'd be grateful for a review for that. It's not yet the way suggested by @jdc91 yet, but I think it's a step forward of decoupling the hidden network dependency :)